app/controllers/project_folders_controller.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class ProjectFoldersController < ApplicationController
+  before_action :check_manage_permissions, only: %i(move_to)
+
+  def move_to
+    destination_folder = current_team.project_folders.find(move_params[:id])
+    destination_folder.transaction do
+      move_projects(destination_folder)
+      move_folders(destination_folder)
+    end
+    respond_to do |format|
+      format.json { render json: { flash: I18n.t('projects.move.success_flash') } }
+    end
+  rescue StandardError => e
+    Rails.logger.error e.message
+    Rails.logger.error e.backtrace.join("\n")
+    respond_to do |format|
+      format.json { render json: { flash: I18n.t('projects.move.error_flash') }, status: :bad_request }
+    end
+  end
+
+  private
+
+  def check_manage_permissions
+    render_403 unless can_update_team?(current_team)
```

Comment:
From the specifications: "Moving of projects and folders is only allowed to Team administrators (and Org admins)."